src/API/Google/AdsAssetGroup.php

```diff
@@ -283,6 +284,69 @@ protected function get_assets( array $asset_groups ): array {
 		return $asset_groups;
 	}
 
+	/**
+	 * Edit an asset group.
+	 *
+	 * @param int   $asset_group_id The asset group ID.
+	 * @param array $data The asset group data.
+	 * @param array $assets A list of assets data.
+	 *
+	 * @return int The asset group ID.
+	 * @throws ExceptionWithResponseData When an ApiException is caught.
+	 */
+	public function edit_asset_group( int $asset_group_id, array $data, array $assets = [] ): int {
+		try {
+			$operations = $this->asset_group_asset->edit_operations( $asset_group_id, $assets );
+
+			// PMax only supports one final URL but it is required to be an array.
+			if ( ! empty( $data['final_url'] ) ) {
+				$data['final_urls'] = [ $data['final_url'] ];
+				unset( $data['final_url'] );
+			}
+
+			if ( ! empty( $data ) ) {
+				$operations[] = $this->edit_operation( $asset_group_id, $data );
+			}
+
+			if ( ! empty( $operations ) ) {
+				$this->client->getGoogleAdsServiceClient()->mutate( $this->options->get_ads_id(), $operations );
+			}
+
+			return $asset_group_id;
```

Comment:
❓ I take as granted that if the client request is not successful it will trow exception?